src/renderer/editor-mosaic.ts

```diff
@@ -1,85 +1,68 @@
 import * as MonacoType from 'monaco-editor';
 import { MosaicDirection, MosaicNode, getLeaves } from 'react-mosaic-component';
-import { action, observable, reaction } from 'mobx';
-
-import { DEFAULT_MOSAIC_ARRANGEMENT } from './constants';
-import {
-  DEFAULT_EDITORS,
-  DefaultEditorId,
-  EditorId,
-  EditorValues,
-} from '../interfaces';
+import { action, computed, observable, reaction } from 'mobx';
+import { EditorId, EditorValues } from '../interfaces';
 
 import {
   compareEditors,
   getEmptyContent,
-  isKnownFile,
+  isSupportedFile,
+  monacoLanguage,
 } from '../utils/editor-utils';
 
 export type Editor = MonacoType.editor.IStandaloneCodeEditor;
 
-/**
- * Create a mosaic arrangement given an array of editor ids.
- *
- * @export
- * @param {Array<EditorId>} input
- * @returns {MosaicNode<EditorId>}
- */
-export function createMosaicArrangement(
-  input: EditorId[],
-  direction: MosaicDirection = 'row',
-): MosaicNode<EditorId> {
-  // Return single editor or undefined.
-  if (input.length < 2) return input[0];
-
-  // This cuts out the first half of input. Input becomes the second half.
-  const secondHalf = [...input];
-  const firstHalf = secondHalf.splice(0, Math.floor(secondHalf.length / 2));
-
-  return {
-    direction,
-    first: createMosaicArrangement(firstHalf, 'column'),
-    second: createMosaicArrangement(secondHalf, 'column'),
-  };
+export enum EditorPresence {
+  /** The file is known to us but we've chosen not to show it, either
+      because the content was boring or because hide() was called.
+      Its contents are cached offscreen. */
+  Hidden,
+
+  /** Space has been allocated for this file in the mosaic but the
+      monaco editor has not mounted in React yet. This is an interim
+      state before the editor is Visible. */
+  Pending,
+
+  /** The file is visible in one of the mosaic's monaco editors */
+  Visible,
 }
 
-export interface EditorBackup {
-  value?: string;
-  model?: MonacoType.editor.ITextModel | null;
+interface EditorBackup {
+  model: MonacoType.editor.ITextModel;
   viewState?: MonacoType.editor.ICodeEditorViewState | null;
 }
 
 export class EditorMosaic {
-  @observable public readonly editors: Map<EditorId, Editor> = new Map();
   @observable public customMosaics: EditorId[] = [];
```

Comment:
Does it make sense to turn `customMosaics` into a Set given that we don't allow duplicate file names?

Reply:
Probably doesn't matter -- It's going away altogether in the next PR

As of this PR I think it's only read by tests. Another case of "would this change make more sense in `refactor/merge-custom-and-default-editors`